### ISSUE TITLE
Increasing token limit in rust-analyzer

### ISSUE THREAD

**Issue by suhr** [rust-analyzer/rust-analyzer#10855]
I get the following error with rust-analyzer:

```
macro invocation exceeds token limit: produced 997309 tokens, limit is 524288
```

But it's a working code which rustc can compile just fine. So I would like to increase the token limit for rust-analyzer.

**Comment by Veykril** [comment]
Duplicate of https://github.com/rust-analyzer/rust-analyzer/issues/8640

**Comment by WaffleLapkin** [comment]
@Veykril are you sure this is a dup of #8640? Recursion and token limits seem distinct.

**Comment by Veykril** [comment]
Good point, those are indeed different(I do think we had one for token expansion limit as well though)

**Comment by PureWhiteWu** [comment]
Hi, we encountered similar problem. Is there any plan to make this limit configurable?
Thanks!

**Comment by Veykril** [comment]
The problem with makingthis configurable is that ideally this would be configured via an attribute, but rustc has no equivalent attribute for this. Doing this via our plain config isn't really great either though, as that means downstream users of your crate need to set that config as well to properly use your crate

**Comment by PureWhiteWu** [comment]
Hi @Veykril thanks very much for your quick response!
It seems that this limit is hard coded by r-a instead of rustc(rustc doesn't have this limit), so I don't think rustc will provide an equivalent attribute for this, the only way is to add a plain config in r-a.
We encountered this problem as end user, and I think if we can have a way to increase the limit to make things work, that's good enough. As a saying goes "done is better than perfect".

**Comment by ActuallyHappening** [comment]
Would be great to have this, I was using `include_json!` and hit this limit on a 5.2MB file

**Comment by ChayimFriedman2** [comment]
Theoretically we can put it in an attribute under the `rust_analyzer` namespace.

**Comment by teor2345** [comment]
There is a PR to add the rustc attribute here: https://github.com/rust-lang/rust/pull/154968
